src/Configuration/Config.AzureTableStorage/src/ConfigurationTableStore.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.RetryPolicies;
+using Microsoft.WindowsAzure.Storage.Table;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Configuration.AzureTableStorage
+{
+    internal class ConfigurationTableStore : ITableStore<ConfigurationEntry>
+    {
+        private const int RetryBackOffSeconds = 1;
+        private const int RetryMaxAttempts = 3;
+        private readonly CloudTableClient cloudTableClient;
+
+        public ConfigurationTableStore(string connectionString)
+        {
+            if (string.IsNullOrWhiteSpace(connectionString))
+            {
+                throw new ArgumentException($"{nameof(connectionString)} cannot be null or white space.", nameof(connectionString));
+            }
+
+            // Parse storage account
+            var cloudStorageAccount = CloudStorageAccount.Parse(connectionString);
+            var requestOptions = new TableRequestOptions
+            {
+                RetryPolicy = new ExponentialRetry(TimeSpan.FromSeconds(RetryBackOffSeconds), RetryMaxAttempts)
+            };
+
+
+            // Create table client
+            cloudTableClient = cloudStorageAccount.CreateCloudTableClient();
+            cloudTableClient.DefaultRequestOptions = requestOptions;
+        }
+
+        public async Task<IEnumerable<ConfigurationEntry>> GetAllByPartitionKey(string tableName, string partitionKey)
+        {
+            if (string.IsNullOrWhiteSpace(tableName))
```

Comment:
It's an internal class and `tableName` and `partitionKey` are already verifyed in the callee of this method. So these checks could be omitted (maybe just `Debug.Assert` them).
On the other side it is more safe to validate them. I don't have any strong opinion on this.

Reply:
No strong opinion from me on it either. Keeping it as is.